src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1166,24 +1166,12 @@ private static void SortAndRemoveDuplicates<T>(List<T> input, Func<T, string> ke
 
                     if (SessionStateUtilities.MatchesAnyWildcardPattern(entry.Key, functionPatterns, false))
                     {
-                        string message;
-
-                        if (entry.Value.CommandType == CommandTypes.Workflow)
-                        {
-                            message = StringUtil.Format(Modules.ExportingWorkflow, entry.Key);
-                            sessionState.ExportedWorkflows.Add((WorkflowInfo)entry.Value);
-                        }
-                        else
-                        {
-                            message = StringUtil.Format(Modules.ExportingFunction, entry.Key);
-                            sessionState.ExportedFunctions.Add(entry.Value);
-                        }
-
+                        sessionState.ExportedFunctions.Add(entry.Value);
+                        string message = StringUtil.Format(Modules.ExportingFunction, entry.Key);
```

Comment:
Not for this PR - but the result of this format call is typically thrown away. It should be possible to check if methods like `WriteVerbose` will do anything *before* formatting, or at least add overloads that do the formatting only after deciding the formatted string is needed.

Reply:
I think it would be the best to add a few overloads to `WriteVerbose`, which take a format string and arguments. So we can defer the construction of `message` until we know for sure we are going to write out verbose messages. What do you think?

Reply:
That might be easiest to adopt.

Best is to use an `if` before the call, but folks tend to not like that style because it's verbose.

Accepting a delegate is also a clean solution, but that can allocate as well, so might not be an improvement.

Reply:
Open #6740 to track this investigation.